### PR TITLE
added insufficient balance url for each env, dark mode fixes

### PR DESCRIPTION
**Pull request by axelrod-blox**

### CODE REVIEW

File: src/lib/utils/envHelper.ts [hunk 1]
```diff
@@ -116,27 +118,41 @@ const fillNetworkData = (network: NetworkDataFromEnvironmentType, networkId: num
...
+  if (currentNetworkIndex && NETWORKS_DATA.length > 1) {
+    const networkId = NETWORKS_DATA[currentNetworkIndex].networkId;
+    return { ...NETWORKS_DATA[currentNetworkIndex], ...NETWORK_VARIABLES[networkId] };
+  }
+  if (!currentNetworkIndex && process.env.REACT_APP_FAUCET_PAGE) {
+    const holeskyIndex = NETWORKS_DATA.findIndex((networkData: any) => networkData.networkId === NETWORKS.HOLESKY);
+    return saveNetwork(holeskyIndex);
+  }
+  return saveNetwork(FIRST_NETWORK_INDEX);
+};
```
Comment (meshin-dev) [inline src/lib/utils/envHelper.ts:133]: This code became unreadable and hard to understand. 
Let's refactor it now and comment every code block to see what happens and why

**Review by meshin-dev — APPROVED** [review]

File: src/app/components/applications/Faucet/RequestForSsv/RequestForSsv.tsx [hunk 2]
```diff
@@ -15,83 +15,88 @@ import WalletStore from '~app/common/stores/applications/Faucet/Wallet.store';
...
-            }
-            return;
-        }
+  const requestForSSV = async () => {
+    setError('');
+    setButtonText('Requesting...');
+    applicationStore.setIsLoading(true);
+    const response = await faucetStore.registerNewTransaction();
+    if (!response.status) {
+      if (response.type === 2) {
```
Comment (alexstolr) [inline src/app/components/applications/Faucet/RequestForSsv/RequestForSsv.tsx:46]: I don't think that using number for api responses is clear. I understand that it is due to the implementation registerNewTransaction():

} catch (e: any) {
            return { status: false, type: e.response.data.error.message === 'Reached max transactions per day' ? 1 : 2 };
        }

I think we should have a facuetService which calls faucet urls parses its response and errors. We should have faucet errors as constants FaucetError.MAX_DAILY_TX_REACHED . In registerNewTransaction() we check:
type: e.response.data.error.message === 'Reached max transactions per day'. 

then we setError here on line 50 with same text.

Faucet service isn't critical but please add the error as constant.

Also probably it should be "response.type === 1"?

File: src/app/components/applications/Faucet/RequestForSsv/RequestForSsv.tsx [hunk 3]
```diff
@@ -15,83 +15,88 @@ import WalletStore from '~app/common/stores/applications/Faucet/Wallet.store';
...
-            </Grid>
-            {error && <Grid item xs={12} className={classes.ErrorText}>{error}</Grid>}
-            <HCaptcha
-              ref={captchaRef}
-              onVerify={() => setDisabled(false)}
-              sitekey={String(process.env.REACT_APP_CAPTCHA_KEY)}
+    <BorderScreen
+      blackHeader
+      withoutNavigation
+      header={`SSV Faucet ${currentNetworkName()} Testnet`}
```
Comment (alexstolr) [inline src/app/components/applications/Faucet/RequestForSsv/RequestForSsv.tsx:66]: What happens if an exception is thrown?

File: src/lib/utils/envHelper.ts [hunk 4]
```diff
@@ -116,27 +118,41 @@ const fillNetworkData = (network: NetworkDataFromEnvironmentType, networkId: num
 
 export const NETWORKS_DATA = data ? JSON.parse(data).map((network: NetworkDataFromEnvironmentType) => fillNetworkData(network, network.networkId, network.apiVersion)) : null;
 
+export const getCurrentNetwork = () => {
+  if (!NETWORKS_DATA && !process.env.REACT_APP_DISABLE_NETWORK_DATA_CHECK) throw new Error('Provide network data');
```
Comment (alexstolr) [inline src/lib/utils/envHelper.ts:122]: Please make sure that everyone who calls this function handles the exception

File: src/app/components/applications/Faucet/RequestForSsv/RequestForSsv.tsx [hunk 5]
```diff
@@ -15,83 +15,88 @@ import WalletStore from '~app/common/stores/applications/Faucet/Wallet.store';
...
+  const navigate = useNavigate();
+  const captchaRef = useRef(null);
+  const faucetStore: FaucetStore = stores.Faucet;
+  const walletStore: WalletStore = stores.Wallet;
+  const [error, setError] = useState('');
+  const [disabled, setDisabled] = useState(true);
+  const applicationStore: ApplicationStore = stores.Application;
+  const [buttonText, setButtonText] = useState('Request');
+  const [reachedMaxTransactionPerDay, setReachedMaxTransactionPerDay] = useState(false);
+  const captchaTheme = applicationStore.theme.darkMode ? config.THEMES.DARK : config.THEMES.LIGHT;
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/Faucet/RequestForSsv/RequestForSsv.tsx:32]: Captcha component have theme param which should be 'dark' or 'light' string and not boolean

File: src/app/components/applications/Faucet/RequestForSsv/RequestForSsv.tsx (hunk 2 above)
Comment (axelrod-blox) [reply src/app/components/applications/Faucet/RequestForSsv/RequestForSsv.tsx:46]: Type 1 means that we have reached the maximum number of transactions per day. Type 2 means the faucet is depleted.

**Review by alexstolr — APPROVED** [review]
please make small change to extract the errors from config to its' own file.

File: src/app/common/config/config.ts [hunk 6]
```diff
@@ -174,6 +174,10 @@ const config = {
     DARK: 'dark',
     LIGHT: 'light',
   },
+  FAUCET: {
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/common/config/config.ts]: done

**Pull request merged**